lib/experimental/new_embed.dart

```diff
@@ -513,6 +515,10 @@ class NewEmbed {
     }
   }
 
+  void _resetCode() {
```

Comment:
This is only called in one place, though it could also be called on L379. Consider removing it here and just inlining the call to setContextSources on L195.

Reply:
I decided to call `_resetCode` on line 379 - It's a short method but helps to clarify the intent.

Reply:
SGTM